src/rest/Query.ts

```diff
@@ -215,4 +215,13 @@ export interface IQuery {
    * This is the id of the recommendation interface that generated the query.
    */
   recommendation?: string;
+
+  /**
+   * Specifies if the Search API should perform queries containing no "end user" keywords.
+   * 
+   * End users keywords are present in either the {@link IQuery.q} or {@link IQuery.lq} part of the query.
+   * 
+   * This parameter is normally controlled by {@link SearchInterface.options.allowEmptyQuery} option.
+   */
+  allowNoKeywords?: boolean;
```

Comment:
The name is a little confusing in that it begins with a positive (allow) but then is followed with a negative (No). I think inverting the logic to `allowKeywords` is nicer. If you prefer to keep the negative sense, then perhaps `noKeywordsAllowed` or `noKeywords`.

Reply:
I think I'll go with `allowQueriesWithoutKeywords` then.